File: apps/hrms-salary-advance/src/main/java/com/cre/hrms/salaryadvance/SalaryAdvanceApplication.java
package com.cre.hrms.salaryadvance;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@SpringBootApplication(scanBasePackages = {
    "com.cre.hrms.salaryadvance",
    "com.cre.hrms.common",
    "com.cre.hrms.core",
    "com.cre.hrms.persistence",
    "com.cre.hrms.security",
    "com.cre.hrms.config"
})
public class SalaryAdvanceApplication {
    public static void main(String[] args) {
        SpringApplication.run(SalaryAdvanceApplication.class, args);
    }
}

File: apps/hrms-salary-advance/src/main/kotlin/com/cre/hrms/salaryadvance/SalaryAdvanceApplication.kt
package com.cre.hrms.salaryadvance

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.data.jpa.repository.config.EnableJpaRepositories

@SpringBootApplication
@ComponentScan(basePackages = [
    "com.cre.hrms.salaryadvance",
    "com.cre.hrms.security"
])
@EnableJpaRepositories(basePackages = ["com.cre.hrms.persistence.salaryadvance"])
@EntityScan(basePackages = ["com.cre.hrms.persistence.salaryadvance"])
class SalaryAdvanceApplication

fun main(args: Array<String>) {
    runApplication<SalaryAdvanceApplication>(*args)
}
